src/Rings/mpoly-ideals.jl

```diff
@@ -714,10 +725,9 @@ Computes the radical.
   return is_subset(radical(I), I)
 end
 
-function is_known_to_be_radical(I::Ideal)
-  get_attribute(I, :is_radical, false) && return true
-  get_attribute(I, :is_prime, false) && return true
-  return false
+function is_known(::typeof(is_radical), I::MPolyIdeal)
+  is_known(is_prime, I) && is_prime(I) && return true
+  return get_attribute(I, :is_radical, false)
```

Comment:
This doesn't seem right to me: `is_known(is_radical, I)` should return true if the `is_radical` is present -- even if it is set to `false`.
```suggestion
  return has_attribute(I, :is_radical)
```